mkt/extensions/validation.py

```diff
@@ -217,25 +228,45 @@ def validate_icons(self, manifest_json):
 
         * Ensure that, if the icons property is present, a 128px icon is
           provided.
-        * Ensure that each icon referenced has a .png extension.
+        * Ensure that each icon size is a valid integer.
+        * Ensure that each icon file is valid by calling _validate_icon_file().
         """
         icons = manifest_json.get('icons', {})
         if icons:
             if '128' not in icons:
                 self.error('ICONS_NO_128')
-            for size, path in icons.iteritems():
-                if not path.endswith('.png'):
-                    self.error('ICONS_INVALID_FORMAT')
+            for icon_size, icon_path in icons.iteritems():
+                try:
+                    icon_size = int(icon_size)
+                    if icon_size <= 0:
+                        raise ValueError
+                except ValueError:
+                    self.error('ICON_INVALID_SIZE', error_parameter=icon_path)
+                self._validate_icon_file(icon_path, icon_size)
 
-    def validate_icon_files(self, manifest_json):
+    def _validate_icon_file(self, icon_path, icon_size):
         """
-        In turn, validate each icon file referenced by the `icons` property of
-        the manifest:
+        Validate a specific icon path referenced in the manifest:
 
         * Ensure that the file exists in the zip.
         * Ensure that it is a valid PNG file.
-        * Ensure that it has the claimed dimensions.
         * Ensure that it is square.
+        * Ensure that it has the claimed dimensions.
         """
-        icons = manifest_json.get('icons', {})
-        return icons
+        try:
+            icon_contents = self.zipfile.extract_path(icon_path.lstrip('/'))
+        except KeyError:
+            self.error('ICON_DOES_NOT_EXIST', error_parameter=icon_path)
+            return
+        try:
+            image = Image.open(StringIO(icon_contents))
+            if image.format != 'PNG':
```

Comment:
I'm not sure we need to. According to Pillow docs: 
http://pillow.readthedocs.org/en/3.0.x/reference/Image.html#PIL.Image.open
> This is a lazy operation; this function identifies the file, but the file remains open and the actual image data is not read from the file until you try to process the data

I wonder if accessing `format` and `size` count as really reading image data. I'll look into that, if it does some heavy reads then yeah I'll probably need to use `imghdr`.